packages/minimal/src/index.ts

```diff
@@ -177,12 +177,22 @@ export function _callOnClient(method: string, ...args: any[]): void {
 }
 
 /**
- * Starts a Transaction. This is the entry point to do manual tracing. You can
- * add child spans to transactions. Spans themselves can have children, building
- * a tree structure. This function returns a Transaction and you need to keep
- * track of the instance yourself. When you call `.finish()` on the transaction
- * it will be sent to Sentry.
+ * Starts a new `Transaction` and returns it. This is the entry point to manual
+ * tracing instrumentation.
+ *
+ * A tree structure can be built by adding child spans to the transaction, and
+ * child spans to other spans. To start a new child span within the transaction
+ * or any span, call the respective `.startChild()` method.
+ *
+ * Every child span must be finished before the transaction is finished,
+ * otherwise the unfinished spans are discarded.
+ *
+ * The transaction must be finished with a call to its `.finish()` method, at
+ * which point the transaction with all its finished child spans will be sent to
+ * Sentry.
+ *
+ * @param context Properties of the new `Transaction`.
  */
 export function startTransaction(context: TransactionContext): Transaction {
-  return callOnHub('startTransaction', { ...context });
+  return callOnHub('startTransaction', context);
```

Comment:
I would prefer to keep ` { ...context }` with that we don't need to have any undefined checks and we always at least have an empty object to deal with.
If someone in JS would call `Sentry.startTransaction()` this would break in that case.

Reply:
What's up with

https://github.com/getsentry/sentry-javascript/blob/4e552ec0dc7d96a249d0f0ab7df7f4796623a521/packages/hub/src/hub.ts#L368-L380

Reply:
I talked about this with @kamilogorek before changing it, and the conclusion was no shallow copy necessary/desired.

Documented in the commit message.

```
The copying is unnecessary as it is done downstream when creating the
transaction.

This also aligns with the implementation in packages/hub/src/hub.ts.
```

Reply:
Note: I don't have a strong opinion here, but I do appreciate consistency. So if we decide on `{ ...context }` here, I would rather take the same careful approach in similar places.

Reply:
We need to get started somewhere, then I would say let's do it in both `startSpan` and `startTransaction` in the Hub.

Reply:
I retract my statement, my bad. Daniel is right with this statement:
> we always at least have an empty object to deal with.

It seems that `startTransaction` in `hubextensions.ts` file would indeed break when `new Transaction` is called, as this would throw when no argument would be passed to it:

```
if (transactionContext.name) {
  this.name = transactionContext.name;
}
```

Reply:
We should add an empty test that checks `expect(new Transaction()).not.toThrow()` or similar.

Reply:
To avoid controversy here in this PR I've removed this part of the patch / commit.